### ISSUE TITLE
Copy is detected as unused variable

### ISSUE THREAD

**Issue by BernieWhite** [microsoft/PSRule.Rules.Azure#569]
**Description of the issue**

When a top level `copy` is used within the `variables` section of a template, template structure analysis may incorrectly flag this a unused variable.

**To Reproduce**

Steps to reproduce the issue:

```json
{
    "variables": {
        "copy": [
            {
                "name": "solutions",
                "count": "[length(parameters('solutions'))]",
                "input": {
                    "name": "[concat(parameters('solutions')[copyIndex('solutions')].name, '(', parameters('workspaceName'), ')')]",
                    "product": "[concat('OMSGallery/', parameters('solutions')[copyIndex('solutions')].name)]",
                    "promotionCode": "",
                    "publisher": "Microsoft"
                }
            }
        ]
    }
}
```

**Expected behaviour**

The `copy` property should not be flagged unused, properties under the copy should be flagged as unused if they are not.

**Module in use and version:**

- Module: PSRule.Rules.Azure
- Version: **0.18.0-B2011005**

Captured output from `$PSVersionTable`:

```text
Name                           Value
----                           -----
PSVersion                      7.1.0
PSEdition                      Core
GitCommitId                    7.1.0
OS                             Microsoft Windows 10.0.19042
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```
